core/src/main/scala/com/cloudera/spark/CGroupsMemoryHandle.scala

```diff
@@ -0,0 +1,43 @@
+package com.cloudera.spark
+
+import com.cloudera.spark.CGroupsMemoryHandle.cGroupMemoryStatPath
+
+import scala.io.Source
+
+class CGroupsMemoryHandle(cgroupMemStats: Seq[(String, Long)]) extends MemoryGetter {
+
+  override val namesAndReporting: Seq[(String, PeakReporting)] = cgroupMemStats.map(x =>
+    (x._1, IncrementBytes)
+  )
+
+  override def values(dest: Array[Long], offset: Int): Unit = {
+    var counter = 0
+    Source.fromFile(cGroupMemoryStatPath).getLines
+      .map { line =>
+        val words = line.split(" ")
+        words{0} -> words{1}.toLong
+      }.toSeq.foreach{ case (_, metricValue) =>
+      dest(offset + counter) = metricValue
```

Comment:
are you sure the order of the entries in the file never changes?

if we're not sure, this could get the metrics all muddled up.

Lemme explain a bit by describing how and why this plugin works the way it does.  It would be much simpler if each `MemoryGetter` returned a `Map[String, Long]` of metric to value every time its polling for metrics.  But I was worried about the performance impact of really rapid polling, and wanted to avoid things like creating new objects and iterating over collections of objects (which may not be cache-friendly), to prefer iterating on primitive arrays as much as possible.

The way it does this is that each plugin declares its set of metrics once, and it can never change the set of metrics after that (this is in `namesAndReporting`).  then in each call to `values`, the metrics are just written directly into the destination array.  This allows the array to be reused, and late code just iterates over the array (without really caring about names in the tight loop).  But it requires each metric to get written to a consistent location in the array.

In this particular case, it only helps a little bit, because you may need a map of all the metrics internally, to ensure you write them to the correct location in the array.  so I think you might need to first create a map from `name -> position` in the initialization, and then here, look up the name in that map to decide which position to write the value into.

does that make sense?

Reply:
In the latest commit I hard coded the metrics that are supposed to be IncrementCounts instead of IncrementBytes. Originally I hardcoded all the metrics in a specific order to ensure that the metrics are always presented in the same order. This however made the code a little cumbersome. I found the following documentation [memory](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/resource_management_guide/sec-memory) The metrics in my tests are in the exact order as specified by this documentation. This led me to believe that the order of files within an application will not change while the application is running. 

However I share your concern about the metrics being muddled up if we are not sure about the order. But for now I think it is ok to proceed with this code. Please let me know what do you think.